### ISSUE TITLE
No libvlc_set_fullscreen support?

### ISSUE THREAD

**Issue by LinoBarreca** [ZeBobo5/Vlc.DotNet#44]
Hi there :)
Dow do I send a video to full screen?
I tried to search for libvlc_set_fullscreen documented here: https://www.videolan.org/developers/vlc/doc/doxygen/html/group__libvlc__video.html#ga26892692dcb079743c9c7e0df3308ea4 but I didn't find it...

So I tried to VlcManager.SetMediaPlayerVideoHostHandle (mpi, Screen.AllScreens[1].hmonitor)
....but it doesn't work.
It creates a "VLC (hardware YUV DirectX output)" window and plays the video in there.
I'm trying to make it render full screen, without windows borders, captions and so on.
Any hints or workarounds known?

**Comment by sepehr1014** [comment]
Hi, a simple workaround would be to set your window state to none and hide everything once in full screen mode. But that doesn't seem right to me.

**Comment by LinoBarreca** [comment]
I'm trying to implement it adding the functions to Vlc.DotNet.Core.Interops and relative property into Vlc.DotNet.Core.VideoManagement :)

I will let you know if I succeed :)

**Comment by LinoBarreca** [comment]
OK. It can be done this way (if zeBobo5 merges it in the code it needs a bit of clean up but it works perfectly)

in vlcControl.cs I added:
```
        public void GoFullScreen()
        {
            var S = Screen.FromHandle(Handle); //to find the screen the control in on
            var fi = S.GetType().GetField("hmonitor", System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance);
            IntPtr screenhandle = (IntPtr)fi.GetValue(S); //to get the screen handle
            myVlcMediaPlayer.VideoHostControlHandle = screenhandle; //set video as NOT embedded (or full screen doesn't work)
            myVlcMediaPlayer.Video.FullScreen = true; //added property.
        }
```

in IVideoManagement.cs I added:
`        bool FullScreen { set; } //could have get too if needed.`

in VideoManagement.cs I added:
```
        public bool FullScreen
        {
            set
            {
            	myManager.SetFullScreen(myMediaPlayer, value); //set full screen mode
            }
        }
```
Added file \Vlc.DotNet.Core.Interops\VlcManager.SetFullScreen.cs containing:
```
using System;
using Vlc.DotNet.Core.Interops.Signatures;

namespace Vlc.DotNet.Core.Interops
{
    public sealed partial class VlcManager
    {
        public void SetFullScreen(VlcMediaPlayerInstance mediaPlayerInstance, bool fullScreen)
        {
            if (mediaPlayerInstance == IntPtr.Zero)
                throw new ArgumentException("Media player instance is not initialized.");
            GetInteropDelegate<SetFullScreen>().Invoke(mediaPlayerInstance, fullScreen ? 1 : 0);
        }
    }
}
```
Added file \Vlc.DotNet.Core.Interops\Signatures\libvlc_media_player.h\libvlc_set_fullscreen.cs containing:
```
using System;
using System.Runtime.InteropServices;

namespace Vlc.DotNet.Core.Interops.Signatures
{
    /// <summary>
    /// Set fullscreen.
    /// </summary>
    [LibVlcFunction("libvlc_set_fullscreen")]
    [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
    internal delegate void SetFullScreen(IntPtr mediaPlayerInstance, int value);
}
```
and that's it.


How to use it:
open sample.
add a "Fullscreen" button.
on button click event handler just call
`myVlcControl.GoFullScreen();`


HOPE THIS WILL HELP SOMEONE :+1: :)

**Comment by sepehr1014** [comment]
Nice work! How about a "revert to normal mode"? Maybe implement a property "IsFullScreen" ...

**Comment by LinoBarreca** [comment]
yeah sepehr1014, that was just a very quick (1 hour) and dirty (but proper) fix for the problem.
A lot of work can be done to polish it, for example you could add 
```
        public void RevertToWindow()
        {
            myVlcMediaPlayer.Video.FullScreen = false;
            myVlcMediaPlayer.VideoHostControlHandle = Handle; //set video as embedded into control hwnd again
        }
```

or implement, in the same way, the support for `libvlc_get_fullscreen` (to provide a getter for `public bool FullScreen`) and `libvlc_toggle_fullscreen` ....imagination is up to you :)

**Comment by NickStrupat** [comment]
Did this get merged?

**Comment by isedwards** [comment]
@ZeBobo5 - it looks like the code that @LinoBarreca [wrote above](https://github.com/ZeBobo5/Vlc.DotNet/issues/44#issuecomment-93967821) to add a fullscreen option to Vlc.DotNet was never submitted as a pull request (and then the issue was closed and forgotten).

Are either of you able to get this into the codebase, or do you want some help putting it into a PR?

**Comment by wesoos** [comment]
@jeremyVignelles Hi, did this ever get into the code base?

**Comment by jeremyVignelles** [comment]
Hi, never heard of it in Vlc.DotNet, so I guess not. is that something you want to be merged? what would be the difference with setting your own window as fullscreen.
PR are welcome, but be careful, the warning in the documentation says that the child window needs to be reparented.

**Comment by github-wujianwen** [comment]
fullscreen() in winform does not work

**Comment by jeremyVignelles** [comment]
As said here, the binding for the libvlc fullscreen function was added, but it needs reparenting the window, and it's not easy to work with. why not putting your WinForms control in fullscreen as you would normally do instead?
